### PR TITLE
coreboot-toolchain: partially unbreak by making Ada support optional

### LINKED ISSUE #174982: Coreboot Toolchain fails to Build (gnat broken?)
### Steps To Reproduce

1. `nix-build -A pkgs.coreboot-toolchain.i386`

This will result in `pkgs.gnat6` to fail to build.

### Build log

```
...
make[3]: Entering directory '/build/build/libdecnumber'
source='../../gcc-6.5.0/libdecnumber/decNumber.c' object='decNumber.o' libtool=no gcc  -I../../gcc-6.5.0/libdecnumber -I.  -g -W -Wall -Wwrite-strings -Wstrict-prototypes -Wmissing-prototypes -Wold-style-definition -Wmissing-format-attribute -Wcast-qual -pedantic -Wno-long-long  -I../../gcc-6.5.0/libdecnumber -I.   -c ../../gcc-6.5.0/libdecnumber/decNumber.c
In file included from /nix/store/a0s1vq35j3g6aa757v7yg0a001l01ng4-glibc-2.34-210-dev/include/string.h:519,
                 from ../../gcc-6.5.0/libdecnumber/decNumber.c:177:
/nix/store/a0s1vq35j3g6aa757v7yg0a001l01ng4-glibc-2.34-210-dev/include/bits/string_fortified.h:110:50: error: macro "__fortified_attr_access" requires 3 arguments, but only 1 given
In file included from /nix/store/a0s1vq35j3g6aa757v7yg0a001l01ng4-glibc-2.34-210-dev/include/string.h:519,
                 from ../../gcc-6.5.0/libdecnumber/decNumber.c:177:
/nix/store/a0s1vq35j3g6aa757v7yg0a001l01ng4-glibc-2.34-210-dev/include/bits/string_fortified.h:110: error: expected ',' or ';' before '__fortified_attr_access'
make[3]: *** [Makefile:191: decNumber.o] Error 1
make[3]: Leaving directory '/build/build/libdecnumber'
make[2]: *** [Makefile:8706: all-stage1-libdecnumber] Error 2
make[2]: Leaving directory '/build/build'
make[1]: *** [Makefile:21252: stage1-bubble] Error 2
make[1]: Leaving directory '/build/build'
make: *** [Makefile:21573: bootstrap] Error 2
error: builder for '/nix/store/983dhfb06fds2x1yp8zi060ybqgbshmf-gnat-6.5.0.drv' failed with exit code 2;
...
```

Full log: https://gist.github.com/blitz/ee604e1ddf51c9c8e91fe3c5edb36032

### Additional context

I'm currently bisecting this, but it will take a bit. First bad commit is 55c4a726fe944d4a637affde93b2594f0bc0aa07, last good is 5ee3e9f434e1cb1263c331e95d0b3d3bb384e3ec.

### Notify maintainers

@felixsinger 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.17.9, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4`
 - channels(root): `"nixos-21.11.337659.06db2e21974, nixos-hardware, nixos-unstable-22.11pre380124.5ce6597eca7, tuxedo"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by blitz**
###### Description of changes

The coreboot toolchain is broken, because of GNAT 11 failing to build (#186482).  To enable coreboot-toolchain for users that don't require Ada support, I've made the Ada support optional for now.

The default is still to built with Ada support to not surprise users that relied on it.

Relates to #174982.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by blitz** [comment]
```sh
❯ nixpkgs-review pr 187068    
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/187068/head:refs/nixpkgs-review/1
remote: Enumerating objects: 34, done.
remote: Counting objects: 100% (28/28), done.
remote: Compressing objects: 100% (12/12), done.
remote: Total 34 (delta 22), reused 16 (delta 16), pack-reused 6
Unpacking objects: 100% (34/34), 21.86 KiB | 573.00 KiB/s, done.
From https://github.com/NixOS/nixpkgs
   ace5332dcb78..cd8bbdd4fdb1  master                -> refs/nixpkgs-review/0
   e208a4002cbb..c1cba9dc19b2  refs/pull/187068/head -> refs/nixpkgs-review/1
$ git worktree add /home/julian/.cache/nixpkgs-review/pr-187068-2/nixpkgs cd8bbdd4fdb15f7758fd3c8df531e9d42bee239d
Preparing worktree (detached HEAD cd8bbdd4fdb1)
Updating files: 100% (31437/31437), done.
HEAD is now at cd8bbdd4fdb1 Merge pull request #187036 from r-ryantm/auto-update/symfony-cli
$ nix-env --option system x86_64-linux -f /home/julian/.cache/nixpkgs-review/pr-187068-2/nixpkgs -qaP --xml --out-path --show-trace
$ git merge --no-commit --no-ff c1cba9dc19b2cf1f475a472aa389712dedd1ae9b
Automatic merge went well; stopped before committing as requested
$ nix-env --option system x86_64-linux -f /home/julian/.cache/nixpkgs-review/pr-187068-2/nixpkgs -qaP --xml --out-path --show-trace --meta
7 packages updated:
coreboot-toolchain-aarch64 coreboot-toolchain-arm coreboot-toolchain-i386 coreboot-toolchain-nds32le coreboot-toolchain-ppc64 coreboot-toolchain-riscv coreboot-toolchain-x64

$ nix --experimental-features nix-command build --no-link --keep-going --option build-use-sandbox relaxed -f /home/julian/.cache/nixpkgs-review/pr-187068-2/build.nix

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/187068

7 packages built:
coreboot-toolchain.aarch64 coreboot-toolchain.arm coreboot-toolchain.i386 coreboot-toolchain.nds32le coreboot-toolchain.ppc64 coreboot-toolchain.riscv coreboot-toolchain.x64

$ /nix/store/d98zwyxdbxldn15ccrims9z0jdknhydf-nix-2.10.3/bin/nix-shell /home/julian/.cache/nixpkgs-review/pr-187068-2/shell.nix

```

**Comment by felixsinger** [comment]
I will look into this.

**Comment by blitz** [comment]
> It builds just fine using `nix-shell --pure` and https://review.coreboot.org/c/coreboot/+/66835
> 
> I applied the first two of your patches locally on nixpkgs master. The [build.log](https://github.com/NixOS/nixpkgs/files/9364545/build.log). Not sure what is wrong here.

From looking at [other people](https://github.com/ghdl/ghdl/pull/2057/commits/8c5689b93e68ac34cdf7dd089a3990ae22f0049c), it seems this is actually a warning turned into an error by `-Werror`?

**Comment by blitz** [comment]
GCC seems to have fixed the offending code: https://github.com/gcc-mirror/gcc/commit/72a29376c63172540576bd9b1d20f5c7c0e42cf3#diff-19b4c901fc006983ab2e0f00669b2616f47ceb57105ff505a1c8df5997e1c187

**Comment by blitz** [comment]
Mmh. I tried to disable warnings with `-gnatwA`, but it doesn't help. Weird. Patching the code seems not really feasible, because it's complaining at just too many places.

**Comment by felixsinger** [comment]
I'm just looking into updating GCC from 11.2 to 12.1.

**Comment by felixsinger** [comment]
I created a patch to update GCC from 11.2 to 12.2 https://review.coreboot.org/c/coreboot/+/66922. I will create a test PR here later to see if it builds with Hydra.

File: pkgs/development/tools/misc/coreboot-toolchain/default.nix [hunk 1]
```diff
@@ -8,11 +8,13 @@ let
...
     , getopt
     , git
-    , gnat11
+    , gnat12
+    , gcc
     , lib
     , perl
     , stdenvNoCC
     , zlib
+    , enableAda ? false
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/coreboot-toolchain/default.nix:17]:
```suggestion
    , withAda ? false
```

Comment (blitz) [reply pkgs/development/tools/misc/coreboot-toolchain/default.nix]: Fixed. Thanks!

**Comment by blitz** [comment]
@felixsinger I've dropped the GCC/GNAT 12 update from this PR. I think it would still be worthwhile to be able to flip Ada support off for people that don't need it. I can flip the default of the `withAda` flag to true to not create surprises for existing users. What do you say?

**Comment by SuperSandro2000** [comment]
> I've rebased this PR, all the coreboot-toolchain derivations are still building:

The interesting part of nixpkgs-review is the ``report.md`` which can be posted with ``nixpkgs-review post-result``.

**Review by SuperSandro2000 — APPROVED** [review]

**Review by ghost — APPROVED** [review]
I submitted a very similar change in #173808, and still have it in my local tree because gnat is such a headache.

**Comment by felixsinger** [comment]
> I can flip the default of the `withAda` flag to true to not create surprises for existing users. What do you say?

Yes, please do that.

File: pkgs/development/tools/misc/coreboot-toolchain/default.nix [hunk 2]
```diff
@@ -9,10 +9,12 @@ let
     , getopt
     , git
     , gnat11
+    , gcc
     , lib
     , perl
     , stdenvNoCC
     , zlib
+    , withAda ? false
```
Comment (felixsinger) [inline pkgs/development/tools/misc/coreboot-toolchain/default.nix:17]: Change default to true.

**Review by felixsinger — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/misc/coreboot-toolchain/default.nix (hunk 2 above)
Comment (blitz) [reply pkgs/development/tools/misc/coreboot-toolchain/default.nix]: Done.

**Comment by blitz** [comment]
> > I can flip the default of the `withAda` flag to true to not create surprises for existing users. What do you say?
> 
> Yes, please do that.

I flipped the default.

**Review by felixsinger — APPROVED** [review]
Looks good. Thank you!

**Pull request merged**